File: largest-number-at-least-twice-of-others.ts
{
    // todo 
    function dominantIndex(nums: number[]): number {
        const max = Math.max(...nums);
        const diff = max / 2;
        let search;

        for (let i = 0; i < nums.length; i++) {
            const num = nums[i];

            if (num === max) {
                search = i;
                continue;
            }

            if (diff < num) {
                return -1;
            }
        }

        return search;

    };

    console.log(dominantIndex([3,6,1,0]))
}

File: largest-number-at-least-twice-of-others.rs
pub fn dominant_index(nums: Vec<i32>) -> i32 {
    let mut search_index = 0;
    let max_value = nums.iter().max().unwrap();
    let diff = max_value / 2;

    for i in 0..nums.len() {
        let num = nums[i];

        if num == *max_value {
            search_index = i;
            continue;
        }

        if num > diff {
            return -1;
        }


    }

    return search_index as i32;
}